work25/register.js

```diff
@@ -0,0 +1,183 @@
+// modal関係
+const modal = document.getElementById('js-modal')
+const mask = document.getElementById('js-mask')
+const checkbox = document.getElementById('js-checkbox')
+const agreeText = document.getElementById('js-agreeText')
+const closeIcon = document.getElementById('js-closeIcon')
+const closeBtn = document.getElementById('js-closeBtn')
+const submitBtn = document.getElementById('js-submitBtn')
+
+// form関係
+const userName = document.getElementById('userName')
+const mail = document.getElementById('mail')
+const password = document.getElementById('password')
+
+const flags = {
+  checkbox: false,
+  userName: false,
+  mail: false,
+  password: false
+}
+
+/**
+ * 各フォームのイベント発火タイミング
+ * ユーザ名： 入力時
+ * メールアドレス： フォームからフォーカスが離れた時
+ * パスワード： フォームからフォーカスが離れた時
+ */
+userName.addEventListener('input', () => {
+  const inputUserName = userName.value
+  const result = nameLengthCheck(inputUserName)
+  const userNameErrorMessage = document.getElementById('userNameErrorMessage')
+
+  if (result) {
+    userNameErrorMessage?.classList.remove('active')
+    flags.userName = true
+  } else {
+    outPutErrorMessage(userName, userNameErrorMessage)
+    flags.userName = false
+  }
+
+  submitBtnFlag()
+})
+
+mail.addEventListener('blur', () => {
+  const inputMail = mail.value
+  const result = mailCheck(inputMail)
+  const mailErrorMessage = document.getElementById('mailErrorMessage')
+
+  if (result) {
+    mailErrorMessage?.classList.remove('active')
+    flags.mail = true
+  } else {
+    outPutErrorMessage(mail, mailErrorMessage)
+    flags.mail = false
+  }
+
+  submitBtnFlag()
+})
+
+password.addEventListener('blur', () => {
+  const inputPassword = password.value
+  const result = passwordCheck(inputPassword)
+  const passwordErrorMessage = document.getElementById('passwordErrorMessage')
+
+  if (result) {
+    passwordErrorMessage?.classList.remove('active')
+    flags.password = true
+  } else {
+    outPutErrorMessage(password, passwordErrorMessage)
+    flags.password = false
+  }
+
+  submitBtnFlag()
+})
+
+/**
+ * 各フォームのバリデーション
+ * ユーザ名： 
+    - 1文字以上、16文字未満
+ * メールアドレス： softbankを参考（https://www.softbank.jp/support/faq/view/10544）
+    - ひとつ目の文字は、アルファベットまたは数字のみ
+    - 文字数は、3文字以上30文字以内
+    - 半角英小文字、数字、「－」（ハイフン）、「．」（ドット）、「＿」（アンダーバー）のみ
+ * パスワード： 
+    - 8文字以上の大小の英数字を交ぜたもの
+ */
+const nameLength = {
+  minLength: 1,
+  maxLength: 15
+}
+const nameLengthCheck = (inputUserName) => {
+  return inputUserName.length >= nameLength.minLength && inputUserName.length <= nameLength.maxLength
+}
+
+const mailRegex = /^[a-z\d][0-9a-z\.-]{2,29}@.+\..{2,}/
```

Comment:
```suggestion
const mailSpecification = {
   regex:  /^[a-z\d][0-9a-z\.-]{2,29}@.+\..{2,}/
}
```

ただの変数だけだと他の仕様が入ってくる余地がないです